core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/AggregationDecorrelation.java

```diff
@@ -51,4 +53,20 @@ public static Map<Symbol, Aggregation> rewriteWithMasks(Map<Symbol, Aggregation>
 
         return rewritten.buildOrThrow();
     }
+
+    public static AggregationNode restoreDistinctAggregation(
```

Comment:
please find a better name for it or add some comment. It's hard to tell what `restoreDistinctAggregation` is about from the name